### PR TITLE
Add more Build service secrets lookup options

### PR DESCRIPTION
**Pull request by mshaposhnik**
Add some more options for the SCM secrets lookup in bild service.

### CODE REVIEW

File: docs/modules/ROOT/pages/how-tos/configuring/creating-secrets.adoc [hunk 1]
```diff
@@ -67,9 +67,55 @@ Using the PAT authentication requires only the `password` key. The `username` sh
 
 This secret will be used by the build service to perform both default and Pipeline-as-Code builds.
 
+
+It is also possible to have secrets for per-repository or organization access. To do this, a `appstudio.redhat.com/scm.repository` annotation should be added to the secret. It may either specify the full repository path or the
+partial path with a wildcard. For example, to create a secret for all repositories in the `my-user` organization, it should be created as follows:
```
Comment (missmesss) [inline docs/modules/ROOT/pages/how-tos/configuring/creating-secrets.adoc:72]: Lines 71 and 72 should be one line.

Comment (missmesss) [inline docs/modules/ROOT/pages/how-tos/configuring/creating-secrets.adoc:72]:
```suggestion
partial path with a wildcard. For example, to create a secret for all repositories in the `my-user` organization, create (or add) the following YAML file:
```

I mean that we need an imperative-based phrase "to create a secret..., do this..."

File: docs/modules/ROOT/pages/how-tos/configuring/creating-secrets.adoc [hunk 2]
```diff
@@ -67,9 +67,55 @@ Using the PAT authentication requires only the `password` key. The `username` sh
...
+  annotations:
+    appstudio.redhat.com/scm.repository: <repository-path> # for example, my-user/my-repo
+type: kubernetes.io/basic-auth
+stringData:
+  password: <PERSONAL GITLAB ACCESS TOKEN>
+----
+
+[NOTE]
+====
+It is also allowed to have multiple repositories listed under the `appstudio.redhat.com/scm.repository` annotation (comma separated). The secret will be used for all repositories that match the specified paths.
```
Comment (missmesss) [inline docs/modules/ROOT/pages/how-tos/configuring/creating-secrets.adoc:113]:
```suggestion
You can have multiple repositories listed under the `appstudio.redhat.com/scm.repository` annotation. Separate repository names with commas when listing them. The secret will be used for all repositories that match the specified paths.
```

We try to keep RHTAP/Konflux docs more user-focused and action-based, so it's preferred to say "you can" or "do this".

File: docs/modules/ROOT/pages/how-tos/configuring/creating-secrets.adoc [hunk 3]
```diff
@@ -67,9 +67,55 @@ Using the PAT authentication requires only the `password` key. The `username` sh
...
+====
+
 [IMPORTANT]
 ==== 
-* Each workspace can contain only one set of credentials for each Source Control Management (SCM) system, such as GitHub or GitLab. It means you can’t add two different tokens for a GitLab repository to a single workspace.
+* Secrets lookup mechanism is searching for the most specific secret first. So the secret with repository annotation will be used first if it matches the component repository path. In none found, then lookup will try to find the one with wildcard, or just the host matching one.
```
Comment (missmesss) [inline docs/modules/ROOT/pages/how-tos/configuring/creating-secrets.adoc:118]:
```suggestion
* Secrets lookup mechanism is searching for the most specific secret first. The secret with a repository annotation will be used first if it matches the component repository path. In none found, then a lookup will try to find a secret with a wildcard, or the host matching one.
```

**Review by missmesss — APPROVED** [review]
Just a few language nitpicks, feel free to keep your wording too.

**Comment by mshaposhnik** [comment]
Thanks for the suggestions @missmesss

**Pull request merged**